src/pydata_sphinx_theme/__init__.py

```diff
@@ -284,7 +284,10 @@ def setup(app: Sphinx) -> Dict[str, str]:
 
     app.add_html_theme("pydata_sphinx_theme", str(theme_path))
 
-    app.add_post_transform(short_link.ShortenLinkTransform)
+
+    theme_options = utils.get_theme_options_dict(app)
+    if theme_options.get("shorten_urls") is True:
+        app.add_post_transform(short_link.ShortenLinkTransform)
```

Comment:
for this change, you'll need to add `shorten_urls` to `src/pydata_sphinx_theme/theme/pydata_sphinx_theme/theme.conf` and give it a default value (probably `True` for backwards compatibility)